lib/jormungandr/exe/cardano-wallet-jormungandr.hs

```diff
@@ -273,7 +276,25 @@ genesisHashOption = optionT $ mempty
     <> help "Blake2b_256 hash of the genesis block, in base 16."
 
 -- | -- [ARGUMENTS...]
-extraArguments :: Parser [Text]
-extraArguments = some $ argument str $ mempty
+extraArguments :: Parser [String]
+extraArguments = many $ argument jmArg $ mempty
     <> metavar "[-- ARGUMENTS...]"
     <> help "Extra arguments to be passed to jormungandr."
+  where
+    jmArg = do
+        arg <- readerAsk
+        case validate arg of
+            Just err -> readerError err
+            Nothing -> pure arg
+    validate arg
+        | "--genesis-block" `isPrefixOf` arg = Just $
+            "The " <> arg <> " option must be placed before the --"
+        | "--rest-listen" `isPrefixOf` arg = Just $
+            suggestion "--rest-listen"
+        | "--storage" `isPrefixOf` arg = Just $
+            suggestion "--storage"
+        | otherwise = Nothing
+    suggestion arg = "The " <> arg <> " argument is used by "
+        <> "`cardano-wallet-jormungandr launch'."
```

Comment:
I think it's OK as is - the meaning is conveyed well enough. But if you really want we can use `unsafePerformIO getProgName`.

Reply:
For consistency, I think it'd be better to hardcode `cardano-wallet` (if hardcoding is better), since it is the name of the released binary. But that's nitpicking...